app/uk/gov/hmrc/individualsbenefitsandcreditsapi/handlers/CustomErrorHandler.scala

```diff
@@ -18,15 +18,56 @@ package uk.gov.hmrc.individualsbenefitsandcreditsapi.handlers
 
 import javax.inject.Inject
 import play.api.Configuration
+import play.api.http.Status.{BAD_REQUEST, NOT_FOUND}
+import play.api.libs.json.Json
+import play.api.mvc.Results.Status
+import play.api.mvc.{RequestHeader, Result}
+import uk.gov.hmrc.individualsbenefitsandcreditsapi.domains.{
+  ErrorInvalidRequest,
+  ErrorNotFound
+}
+import uk.gov.hmrc.play.HeaderCarrierConverter
 import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.bootstrap.backend.http.{ErrorResponse, JsonErrorHandler}
 import uk.gov.hmrc.play.bootstrap.config.HttpAuditEvent
-import uk.gov.hmrc.play.bootstrap.http.JsonErrorHandler
 
 import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
 class CustomErrorHandler @Inject()(auditConnector: AuditConnector,
                                    httpAuditEvent: HttpAuditEvent,
                                    configuration: Configuration)
     extends JsonErrorHandler(auditConnector, httpAuditEvent, configuration) {
-  //TODO - implement at a later date (see individuals-income)
+  import httpAuditEvent.dataEvent
+
+  override def onClientError(request: RequestHeader,
+                             statusCode: Int,
+                             message: String): Future[Result] = {
+
+    implicit val headerCarrier =
+      HeaderCarrierConverter.fromHeadersAndSessionAndRequest(request.headers,
+                                                             request =
+                                                               Some(request))
+
+    statusCode match {
+      case NOT_FOUND =>
+        auditConnector.sendEvent(
```

Comment:
So we do have audit connectors then. They simply cover the error scenarios as it stands. 👍 

We can extend for request and responses.